rotkehlchen/api/rest.py

```diff
@@ -5420,3 +5421,29 @@ def prepare_native_transfer(
             return wrap_in_fail_result(str(e), status_code=HTTPStatus.BAD_REQUEST)
 
         return _wrap_in_ok_result(result=payload)
+
+    @async_api_call()
+    def fetch_token_balance_for_address(
+            self,
+            address: ChecksumEvmAddress,
+            evm_chain: EVM_CHAIN_IDS_WITH_TRANSACTIONS_TYPE,
+            asset: 'CryptoAsset',
+    ) -> dict[str, Any]:
+        node_inquirer = self.rotkehlchen.chains_aggregator.get_evm_manager(evm_chain).node_inquirer
+        try:
+            if asset == node_inquirer.native_token:
+                balance = node_inquirer.get_multi_balance([address])[address]
+            else:
+                balance = asset_normalized_value(
+                    asset=asset,
+                    amount=node_inquirer.call_contract(
+                        contract_address=asset.resolve_to_evm_token().evm_address,
+                        abi=node_inquirer.contracts.erc20_abi,
+                        method_name='balanceOf',
+                        arguments=[address],
+                    ),
+                )
+        except RemoteError as e:
+            return wrap_in_fail_result(str(e), status_code=HTTPStatus.INTERNAL_SERVER_ERROR)
```

Comment:
double check it but I believe we don't raise internal server errors for remoteError.It is 409 conflict irc

Reply:
we raise both across the codebase and even `400 BAD REQUEST` in some scenario.